src/providers/openstack/mod.rs

```diff
@@ -14,4 +14,20 @@
 
 //! openstack metadata fetcher
 
+use crate::errors;
+use crate::providers;
+use configdrive::OpenstackConfigDrive;
+use network::OpenstackProviderNetwork;
+use slog_scope::warn;
+
+pub mod configdrive;
 pub mod network;
+
+pub fn try_config_drive_else_network() -> errors::Result<Box<dyn providers::MetadataProvider>> {
```

Comment:
A short docstring on this would be helpful, in order to mention why we do this serially.